_drafts/2020-07-27-data_science_plotting_notebook_template.md

```diff
@@ -0,0 +1,131 @@
+
+The notebook begins with a bunch of boilerplate to set some metadata that is
+saved to the files and change some defaults like the figure size and
+resolution, font size, and legend frame. After that there are a few helpful
+functions which I will discuss below.
+
+### Draw Bands
+
+One of my favorite functions is `draw_bands()`. It draws a set of colored
+bands on the background of the plot based on the axis tick locations.
+
+When called with just the axis, like `draw_bands(ax)`, it produces this:
+
+[![A plot showing the default grey bands.][bands_plot]][bands_plot]
+
+[bands_plot]: {{ file_dir }}/bands.svg
+
+But you can also customize the color using `draw_bands(ax, color="orange",
+alpha=0.05)`, which produces:
+
+[![A plot showing the orange bands.][orange_bands_plot]][orange_bands_plot]
+
+[orange_bands_plot]: {{ file_dir }}/orange_bands.svg
+
+These bands are a subtle way of indicating where on the X-axis a point lies,
+which is especially useful when plotting a time series. I use them often. Here
+are some examples:
+
+- [**Discussing my sons' language development**][language_post] to highlight each month.
+- [**Plotting the progression of the cycling hour record**][hour_post] to show each decade.  
+- [**Exploring when cyclists are involved in traffic accidents**][bike_post] to highlight the seasonality.
+
+[language_post]: {% post_url 2020-02-10-my_sons_language_development_comparison %}#development
+[hour_post]: {% post_url 2019-07-09-hour_record_plot_improvements %}#improvements
+[bike_post]: {% post_url 2019-02-20-switrs_bicycle_crashes_by_date %}#day-by-day
+
+### Draw Legends
+
+I like minimal, but informative, legends. Color alone is often enough to
+differentiate lines or points on simple plots, so I wrote a function to change
+the color of the legend text to match the line, called
+`draw_colored_legend()`. It produces a legend like on this plot:
+
+[![A plot showing my colored legend.][legend_plot]][legend_plot]
+
+[legend_plot]: {{ file_dir }}/legend.svg
+
+This function is a little [brittle][brittle]; it works well for scatter plots,
+but fails for some other matplotlib objects. I plan to make it more robust in
+the future.
+
+[brittle]: https://en.wikipedia.org/wiki/Software_brittleness
+
+I have used these types of legends when:
+
+- [**Plotting my son's language development**][son_post] to label each language.
+- [**Plotting Tour de France Prize Money**][tdf_post] to label the winner's prize compared to the total.
+- [**Comparing Data Science Salaries by Gender**][salary_post] to differentiate the points for men and women.
+
+[son_post]: {% post_url 2020-01-30-my_second_sons_words %}#the-words
+[tdf_post]: {% post_url 2019-11-25-tdf_prize_money_plot_improvements %}#improvements
+[salary_post]: {% post_url 2019-05-09-data_science_salaries_by_gender %}#by-region
+
+## Putting It Together
+
+The [plotting notebook][plotting_nb] lets you make beautiful plots quickly and
+easily. It lets you make this plot:
```

Comment:
The plotting notebook *allows you to* make...

Otherwise repeated "lets" is clunky